common/txmgr/types/client.go

```diff
@@ -0,0 +1,34 @@
+package types
+
+import (
+	"context"
+	"math/big"
+)
+
+type Client[CHAINID any, ADDR comparable, BLOCK any, HEADER any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
+	ChainID() (CHAINID, error)
+
+	// account
+	BalanceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (*big.Int, error)
+	TokenBalance(ctx context.Context, accountAddress ADDR, tokenAddress ADDR) (*big.Int, error)
+	NonceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (uint64, error)
+
+	// tx
+	SendTransaction(ctx context.Context, tx *TX) error
+	SimulateTransaction(ctx context.Context, tx *TX) error
+	TransactionByHash(ctx context.Context, txHash HASH) (*TX, error)
+	TransactionReceipt(ctx context.Context, txHash HASH) (*TXRECEIPT, error)
+
+	// events
+	FilterLogs(ctx context.Context, query EVENTOPS) ([]EVENT, error)
+
+	// block
+	BlockByNumber(ctx context.Context, number *big.Int) (*BLOCK, error)
+	BlockByHash(ctx context.Context, hash HASH) (*BLOCK, error)
+	LatestBlockHeight(context.Context) (*big.Int, error)
+	HeaderByNumber(context.Context, *big.Int) (*HEADER, error)
+	HeaderByHash(context.Context, HASH) (*HEADER, error)
+
+	// contract
+	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
```

Comment:
Suggestion #2: Rename `CallContext` into `CallContract` and existing `CallContract` into something else.
Calling contract is commonly referenced as reading from the contract without state change (that's what `CallContext` is doing atm)

Reply:
The scope of `CallContract` is specifically designed to make contract calls.`CallContext` can be used for other RPC methods as well, so I think it's fine as is.

Reply:
That's a good point 👍 
Then I think we need both.